compiler/src/iree/compiler/Codegen/LLVMCPU/LLVMCPUTensorPad.cpp

```diff
@@ -108,6 +108,9 @@ void LLVMCPUTensorPadPass::runOnOperation() {
     context->getLoadedDialect<tensor::TensorDialect>()
         ->getCanonicalizationPatterns(patterns);
     tensor::PadOp::getCanonicalizationPatterns(patterns, context);
+    // Fold multiple tensor.extract_slice ops into a single one. They will be
+    // unfolded back later in the pipeline.
+    tensor::populateMergeConsecutiveInsertExtractSlicePatterns(patterns);
```

Comment:
I was not aware of these patterns, they look nice!

Reply:
Neither do I! Nicolas told me. I think it was added recently. There are patterns to merge and split extract slices which address my concerns about loop invariance.